lua/telescope/builtin/init.lua

```diff
@@ -859,6 +859,60 @@ builtin.marks = function(opts)
   }):find()
 end
 
+builtin.registers = function(opts)
+  opts = opts or {}
+
+  local registers_table = {}
+  table.insert(registers_table, {kind = "Unnamed",      key = "\""})
+  table.insert(registers_table, {kind = "Small Delete", key = "_"})
+  table.insert(registers_table, {kind = "Alternate",    key = "#"})
+  table.insert(registers_table, {kind = "Expression",   key = "="})
+  table.insert(registers_table, {kind = "Black Hole",   key = "_"})
+  table.insert(registers_table, {kind = "Last Search",  key = "/"})
+
+  -- clipboaord
+  local selection = {"*", "+"}
+  for i in pairs(selection) do
+    table.insert(registers_table, {kind = "Selection",    key = selection[i]})
+  end
+
+  -- named
+  for i = 0, 9 do
+    table.insert(registers_table, {kind = "Numbered", key = tostring(i)})
+  end
+
+  -- alpha
+  for i = 65, 90 do
+    table.insert(registers_table, {kind = "Named", key = string.char(i)})
+  end
+
+  -- readonly
+  local readonly = {":", ".", "%"}
+  for i in pairs(readonly) do
+    table.insert(registers_table, {kind = "Read-Only",    key = readonly[i]})
+  end
+
+  pickers.new(opts,{
+    prompt_title = 'Registers',
+    finder = finders.new_table {
+      results = registers_table,
+      entry_maker = make_entry.gen_from_registers(opts),
+    },
+    previewer = previewers.vimgrep.new(opts),
+    -- use levenshtein as n-gram doesn't support <2 char matches
+    sorter = sorters.get_levenshtein_sorter(),
+    attach_mappings = function(_, map)
+      -- TODO: Find a way to insert the text... it seems hard.
+      -- map('i', '<C-i>', actions.insert_value, { expr = true })
+      map('i', '<CR>', actions.insert_value)
```

Comment:
You should use `actions.goto_file_selection_edit:replace(actions.insert_value)`. That is a cool new api done by tj. In that way we respect user mappings. Take a look at what other `builtins` do with `replace`

Reply:
I'll look into this  and update tomrrow.

Reply:
To be honestr, that mapping wasn't used and I'm not sure how `:replace` fits in with the other mappings. If it's still applicable, let me know.

Reply:
The thing about `actions.goto_file_selection_edit:replace` is that we basically replacing the users main action.
So if a user doesn't like `<CR>` as mapping and likes to use idk `<space>` he will hit `<space>` and we would call `actions.goto_...`. With that replace we basically just follow what the user expects for all individual pickers, even though they do something differently like checking out a branch or stuff. Does that makes sense.